### PR TITLE
Enhancement: support automatically assigning custom fields via consumption templates

### PR DESCRIPTION
**Pull request by shamoon**
## Proposed change

I think this is relatively straightforward with consumption templates. One note is there was some debate about whether it would be 'ok' to leave the field value as required (since it will be empty when automatically added), but I think overall theres no real downside to changing them to allow null. In thinking about use case(s) I can imagine people would want to add the field but maybe fill in the value later (e.g. its an invoice, add field "paid date", fill it out once its actually paid). But of course let me know if other thoughts

Screenshot 2023-11-30 at 9 52 45 AM

Fixes #4644

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (please explain):

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [x] If applicable, I have included testing coverage for new code in this PR, for [backend](https://docs.paperless-ngx.com/development/#testing) and / or [front-end](https://docs.paperless-ngx.com/development/#testing-and-code-style) changes.
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [x] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [x] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [x] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

File: src/documents/consumer.py [hunk 1]
```diff
@@ -779,6 +789,13 @@ def apply_overrides(self, document):
             }
             set_permissions_for_object(permissions=permissions, object=document)
 
+        if self.override_custom_field_ids:
+            for field_id in self.override_custom_field_ids:
+                field = CustomField.objects.get(pk=field_id)
+                document.custom_fields.add(
```
Comment (stumpylog) [inline src/documents/consumer.py:795]: I don't think you need the `.add` here

Comment (shamoon) [reply src/documents/consumer.py:795]: Sorry if im missing something obvious, why not (or what do you mean instead)? This is adding an instance for each

Comment (stumpylog) [reply src/documents/consumer.py:795]: The custom_fields will be update when a CustomFieldInstance is created linked to that document.  So I don't think you need to do anything with the field on the Document.

Comment (shamoon) [reply src/documents/consumer.py:795]: ahh, yes, duh. Thanks, will fix

**Review by stumpylog — APPROVED** [review]

**Pull request merged**